src/js/gi/components/search/KeywordSearch.jsx

```diff
@@ -15,6 +15,33 @@ export class KeywordSearch extends React.Component {
     this.clickedSuggestionValue = this.clickedSuggestionValue.bind(this);
     this.renderSuggestion = this.renderSuggestion.bind(this);
     this.shouldRenderSuggestions = this.shouldRenderSuggestions.bind(this);
+    this.shouldRenderSuggestions = this.shouldRenderSuggestions.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+    this.onKeyUp = this.onKeyUp.bind(this);
+    this.handleSuggestionSelected = this.handleSuggestionSelected.bind(this);
+  }
+
+  componentDidMount() {
+    const searchQuery = this.props.location && this.props.location.query;
+    if (searchQuery) {
+      this.handleChange(null, { newValue: searchQuery.name, method: 'enter' });
+    }
+  }
+
+  onKeyUp(e) {
+    const { onFilterChange, autocomplete } = this.props;
+    if ((e.which || e.keyCode || 0) === 13) {
+      e.target.blur();
+      onFilterChange('name', autocomplete.searchTerm);
+    }
+  }
+
+  handleChange(event, data) {
+    this.props.updateAutocompleteSearchTerm(data.newValue);
+  }
+
+  handleSuggestionSelected(event, data) {
+    this.props.onFilterChange('name', data.suggestionValue);
```

Comment:
Good catch; I set a no-op as a default prop for this to prevent any errors on the LandingPage.